Reflection/CustomConfigurationDemo/Attributes/ConfigurationManagerConfigurationItemAttribute.cs

```diff
@@ -0,0 +1,9 @@
+﻿using CustomConfigurationDemo.Providers;
+
+namespace CustomConfigurationDemo.Attributes;
+
+public class ConfigurationManagerConfigurationItemAttribute : ConfigurationItemAttribute
+{
+    public ConfigurationManagerConfigurationItemAttribute(string settingName)
+        : base(settingName, typeof(ConfigurationManagerConfigurationProvider)) { }
```

Comment:
How about calling this attribute explicitly, like:
```suggestion
        : base(settingName, new ConfigurationManagerConfigurationProvider()) { }
```

It will reduce extra requirement of invoking Activator class to create instance. You can use covariance to be able to assign derived types for variable, not only `IConfigurationProvider`

Reply:
Thank you for the suggestion. I've made the necessary adjustments by directly providing an instance of ConfigurationManagerConfigurationProvider in the constructor of ConfigurationManagerConfigurationItemAttribute. This change simplifies the instantiation process and eliminates the need for using typeof and Activator.


```
[AttributeUsage(AttributeTargets.Property)]
public abstract class BaseConfigurationItemAttribute : Attribute
{
    public string SettingName { get; set; }

    public IConfigurationProvider ProviderType { get; set; }

    protected BaseConfigurationItemAttribute(string settingName, IConfigurationProvider providerType)
    {
        SettingName = settingName;
        ProviderType = providerType;
    }
}
```